Issues while building NSQ for windows/amd64

I'm seeing the following errors (on Linux/Mac/Windows machines) when I try to build NSQ for Windows:

Steps followed:
```
$ git clone https://github.com/nsqio/nsq $GOPATH/src/github.com/nsqio/nsq
$ cd $GOPATH/src/github.com/nsqio/nsq
$ dep ensure
$ GOOS=windows GOARCH=amd64 make
```

Build errors:
```
$ GOOS=windows GOARCH=amd64 make
go build  -o build/nsqd ./apps/nsqd
# github.com/nsqio/nsq/internal/dirlock
internal/dirlock/dirlock.go:28:8: undefined: syscall.Flock
internal/dirlock/dirlock.go:28:35: undefined: syscall.LOCK_EX
internal/dirlock/dirlock.go:28:51: undefined: syscall.LOCK_NB
internal/dirlock/dirlock.go:37:9: undefined: syscall.Flock
internal/dirlock/dirlock.go:37:38: undefined: syscall.LOCK_UN
internal/dirlock/dirlock_windows.go:5:6: DirLock redeclared in this block
	previous declaration at internal/dirlock/dirlock.go:11:6
internal/dirlock/dirlock_windows.go:9:6: New redeclared in this block
	previous declaration at internal/dirlock/dirlock.go:16:23
internal/dirlock/dirlock_windows.go:15:19: (*DirLock).Lock redeclared in this block
	previous declaration at internal/dirlock/dirlock.go:22:6
internal/dirlock/dirlock_windows.go:19:19: (*DirLock).Unlock redeclared in this block
	previous declaration at internal/dirlock/dirlock.go:35:6
make: *** [build/nsqd] Error 2
```

I tried the same steps on a native Windows machine, but still no success.

--- a/internal/dirlock/dirlock.go
+++ b/internal/dirlock/dirlock.go
@@ -1,4 +1,4 @@
-// +build !windows !illumos
+// +build !windows,!illumos
 
 package dirlock
 

